CHANGELOG.md

```diff
@@ -41,17 +41,27 @@ Jump in to this release, with the new demos and examples:
-  - `StellarGraph`'s internal cache of node adjacencies is now computed lazily [\#1291](https://github.com/stellargraph/stellargraph/pull/1291) and takes into account whether the graph is directed or not [\#1463](https://github.com/stellargraph/stellargraph/pull/1463) (up to 63% smaller, 30× speedup on a reddit dataset with 11.6 million edges), and they now uses the smallest integer type it can [\#1289](https://github.com/stellargraph/stellargraph/pull/1289) (36% smaller on the reddit dataset)
+  - `StellarGraph`'s internal cache of node adjacencies is now computed lazily [\#1291](https://github.com/stellargraph/stellargraph/pull/1291) and takes into account whether the graph is directed or not [\#1463](https://github.com/stellargraph/stellargraph/pull/1463) (up to 63% smaller, 30× speedup on a reddit dataset with 11.6 million edges), and they now use the smallest integer type they can [\#1289](https://github.com/stellargraph/stellargraph/pull/1289) (36% smaller on the reddit dataset)
   - `StellarGraph`'s internal list of source and target nodes are now stored using integer "ilocs" [\#1267](https://github.com/stellargraph/stellargraph/pull/1267), reducing memory use and making some functionality significantly faster (selection of speedups: `StellarGraph.to_adjacency_matrix` 1.6×, GraphSAGE sampling 1.4×, weighted `BiasedRandomWalk` 2.4× [\#1444](https://github.com/stellargraph/stellargraph/pull/1444), `UniformRandomMetaPathWalk` 6× [\#1446](https://github.com/stellargraph/stellargraph/pull/1446))
   - Functions like `graph.node_features()` no longer needs `node_type` specified if `graph` has only one node type (this includes classes like `HinSAGENodeGenerator`, which no longer needs `head_node_type` if there is only one node type) [\#1375](https://github.com/stellargraph/stellargraph/pull/1375)
+- Performance and memory usage improvements in numbers:
+  - Base `StellarGraph` is up to 4.2 x smaller (tested on FB15K)
+  - Base `StellarGraph` construction is 7 x faster (tested on FB15K)
+  - Undirected `StellarGraph` is up to 2.2 x smaller (tested on FB15K)
+  - Directed `StellarGraph` is up to 1.8 x smaller (tested on FB15K)
+  - Directed adjacency matrix construction is 1.6 x faster (tested on synthetic benchmark)
+  - `HinSAGENodeGenerator` is 2.9 x faster (tested on MovieLens)
+  - `Attri2VecNodeGenerator` is 4.3 x faster (tested on CiteSeer)
+  - Weighted `BiasedRandomWalk` is 2.6 x faster (tested on synthetic benchmark)
+  - `UniformRandomMetapathWalk` is 6.6 x faster (tested on synthetic benchmark)
 
 
 [demos-1.0.0]: https://stellargraph.readthedocs.io/en/v1.0.0/demos/index.html
 
 ### Breaking changes
```

Comment:
Oh, one more potential breaking change: https://github.com/stellargraph/stellargraph/pull/1444 goes from collapsing all multi-edges to considering each separately (summing the weights).

Specifically: it previously required all multi edges had the same weight, and only counted one of them when considering where to walk, so a graph like `a --[1]-- b` (twice times), `a --[1]--- c` would be equivalent to the only having the first edge once, and thus a random walk from `a` would have equal chance of going to `b` or `c` (without biases). Now it is equivalent to having an `a --[2]-- b`, and so `b` is twice as likely as `c`.

Worth mentioning?

Reply:
Hmm right, would something like below be clear enough?
```
- `BiasedRandomWalk` now takes multi-edges into consideration instead of collapsing them when traversing the graph.
```
Or could spell it out a bit more:
```
- `BiasedRandomWalk` now takes multi-edges into consideration instead of collapsing them when traversing the graph. It previously required all multi-edges had to same weight and only counted one of them when considering where to walk, but now a multi-edge is equivalent to having an edge whose weight is the sum of the weights of all edges in the multi-edge.
```

Reply:
The first one is good; could potentially expand very slightly

> - `BiasedRandomWalk` now takes multi-edges into consideration (by effectively summing weights) instead of collapsing them when traversing the graph.

Reply:
(I've put in the long form)

Reply:
That works for me 👍